client/blocks/subscription-length-picker/index.jsx

```diff
@@ -83,6 +111,10 @@ export class SubscriptionLengthPicker extends React.Component {
 									) }
 									value={ planSlug }
 									onCheck={ this.handleCheck }
+									shouldShowTax={ shouldShowTax }
+									taxDisplay={
+										shouldShowTax && this.formatTax( taxRate, priceFull, this.props.currencyCode )
```

Comment:
Just like visibility flag of some common components, e.g. `<img>`, `<Button>`, `<div>`, I think visibility shouldn't affect passing data or not. e.g. This would look more conventional to me:
```
shouldShowTax={ shouldShowTax }
taxDisplay={ this.formatTax( ... ) }
```
However, this loses the benefit of not formatting the tax string if not showing, so the other way would be:
```
shouldShowTax={ ... }
taxRate={ ... }
priceFull={ ... }
currencyCode={ ... }
```
and let `<SubscriptionLengthOption>` to handle it underneath, including visibility and formatting the tax string or not.

Does that make sense to you?

Reply:
Yeah... I wasn't thinking about hiding information from the child as much as not having the necessary data. Only components that should show the tax will have a `taxRate`. `formatTax()` short-circuits in this case, so let's go with your first suggestion.

I'm reluctant to push it down into the option itself, because it would need to get smarter to handle it, and it seems complicated enough already just deciding what to show and where to put it. I like the split between the "what" at the picker level and the "where" at the option level.